### PR TITLE
new reusable workflows part of wultra/tasklist#23

### PR DESCRIPTION
**Pull request by zcgandcomp**

### CODE REVIEW

**Review by romanstrobl — CHANGES_REQUESTED** [review]
Unfortunately, this will not work so easily. The `mvn deploy` command in the GitHub action needs to skip tests, because Push Server Maven tests require a running instance of PowerAuth server. See the error in the GitHub actions related to missing `POWERAUTH_ENDPOINT` variable:

```
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'pushServerTestClientFactory': Injection of autowired dependencies failed; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'POWERAUTH_ENDPOINT' in value "${POWERAUTH_ENDPOINT}/powerauth-java-server/rest"
Caused by: java.lang.IllegalArgumentException: Could not resolve placeholder 'POWERAUTH_ENDPOINT' in value "${POWERAUTH_ENDPOINT}/powerauth-java-server/rest"
```

File: .github/workflows/maven-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  workflow_dispatch:
+    branches:
+      - 'develop'
+      - 'master'
+      - 'releases/**'
+      - 'test/ci'
+    inputs:
+      release_type:
+        type: choice
+        description:
```
Comment (romanstrobl) [inline .github/workflows/maven-deploy.yml:13]: Missing description.

**Review by romanstrobl — APPROVED** [review]
The workflow will be usable after merging pull request https://github.com/wultra/powerauth-push-server/pull/497.

File: .github/workflows/maven-deploy.yml (hunk 1 above)
Comment (zcgandcomp) [reply .github/workflows/maven-deploy.yml]: description added

**Comment by zcgandcomp** [comment]
rebased on current develop

**Pull request merged**